content/blog/2018/istio-multicluster/index.md

```diff
@@ -0,0 +1,290 @@
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: b-ingress
+spec:
+  hosts:
+  - ingress.b.remote
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP	
+  resolution: STATIC
+  endpoints:
+  - address: INGRESS_IP_ADDRESS
+    ports:
+      http: 80
+```
+
+The following VirtualService routes all mesh traffic for hosts `*.svc.b.remote` to the ServiceEntry we just defined above. Note that the port values used here can be used to shift the ports traffic is sent on to the ingress (e.g. to allow for egress/ingress proxy pairs that multiplex multiple protocols over the same physical ports). Also note that this route's destination's host exactly matches the host of the ServiceEntry:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: b-virtual-service
+spec:
+  hosts:
+  - "*.svc.b.remote"
+  gateways:
+  - mesh
+  http:
+  - route:
+    - destination:
+        host: ingress.b.remote
+        port:
+          number: 80
+      weight: 100
+```
+
+Finally, since we're creating hostnames out of nothing, and our apps resolve hostnames via DNS before making HTTP calls, we need our CoreDNS server to resolve the custom host names. This config sends all requests with a `remote` suffix to the CoreDNS server we have running [configured to return a fake IP](#appendix-configuring-the-coredns) for all `*.remote`. `CORE_DNS_IP` should be replaced with the CoreDNS service IP:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-dns
+  namespace: kube-system
+data:
+  stubDomains: |
+    {"remote": ["CORE_DNS_IP"]}
+```
+
+## Security
+
+Each Istio running on a cluster can be configured to have an mTLS connections between the Istio control plane components and between the mesh services. Can we also have an end-to-end mTLS between local caller service and a remote callee service? The answer is yes as long as there is a shared root of trust common to both clusters and accessible for CA validation. Having a common root CA for all local generated CAs will enable the mTLS connection between the egress gateway of _Cluster A_ and ingress gateway of _Cluster B_. And vice versa.
+
+Each local Citadel [can be configured](https://istio.io/docs/tasks/security/plugin-ca-cert/) with the common root CA as well as an upstream CA address and the same mechanism is used by Istio to generate and rotate certificates by the local Citadel. The only difference is the generated CAs will have the root CA in their certificates chain.
+
+If necessary the Istio Citadel can also play the role of the root CA and this can be achieved by running the Citadel as a standalone service on a cluster accessible by both clusters.
+
+[Permission to put Shriram's diagram here]
+
```

Comment:
I feel security section was not well explained in this blog, might be better to split into a separate one.